### PR TITLE
Feature/credentials management

### PR DESCRIPTION
**Pull request by nandagopalan**
Adding draft PR

### CODE REVIEW

File: secrets/localstore.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+import (
+	"bytes"
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/gob"
+	"go.nandlabs.io/commons/errutils"
```
Comment (neo7337) [inline secrets/localstore.go:11]: format the imports

File: secrets/localstore.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	return
+}
+
+func (ls *LocalStore) Provider() string {
+	return LocalStoreProvider
+}
+
+func aesEncrypt(key []byte, message string) (encoded string, err error) {
```
Comment (neo7337) [inline secrets/localstore.go:87]: this function is defined above in aes.go, can we reuse the same?

File: secrets/localstore.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+			stream.XORKeyStream(cipherText[aes.BlockSize:], msgBytes)
+			encoded = base64.RawStdEncoding.EncodeToString(cipherText)
+		}
+	}
+
+	//Return string encoded in base64
+	return
+}
+
+func aesDecrypt(key []byte, content string) (decryptedText string, err error) {
```
Comment (neo7337) [inline secrets/localstore.go:109]: resuse the function from aes.go

**Review by neo7337 — CHANGES_REQUESTED** [review]

**Comment by neo7337** [comment]
@nandagopalan added a few comments

File: secrets/localstore.go (hunk 2 above)
Comment (nandagopalan) [reply secrets/localstore.go:87]: this is going away

File: secrets/localstore.go (hunk 3 above)
Comment (nandagopalan) [reply secrets/localstore.go:109]: yes this is going away as mentioned in the issue number #38

**Review by neo7337 — APPROVED** [review]
approved

**Pull request merged**